core/src/main/java/org/apache/accumulo/core/iterators/OrIterator.java

```diff
@@ -39,29 +39,44 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * An iterator that handles "OR" query constructs on the server side. This code has been adapted/merged from Heap and Multi Iterators.
+ * An iterator that provides a sorted-iteration of column qualifiers for a set of column families in a row.
+ * It is important to note that this iterator <em>does not</em> adhere to the contract set forth by the
+ * {@link SortedKeyValueIterator}. It returns Keys in {@code row+colqual} order instead of
+ * {@code row+colfam+colqual} order. This is required for the implementation of this iterator (to work
+ * in conjunction with the {@code IntersectingIterator}) but is a code-smell. This iterator should only
```

Comment:
Yes it was tongue-in-cheek. I had to google "code smell" since I hadn't heard that phrase before but I don't think its offensive.